Feature/game create

thuanvmse161224 opened the pull request.

thuanvmse161224 committed:
feat: Adding Game Create Page
- Snack Error when error
- Snack Success when success
- Havent have Regex check yet

thuanvmse161224 committed:
refactor: Using config.gms_url

thuanvmse161224 committed:
feat: change create into a dialog
+ adding SnackAlert after add / error
+ make create into a dialog
+ remove createGame route
+ adding DialogHeader code from another git branch

thuanvmse161224 committed:
Merge remote-tracking branch 'origin' into feature/gameCreate

dg-panh reviewed (commented):
![image](https://github.com/FPTU-GHub-Capstone/ghub-app/assets/60615097/5bbda252-45f6-4cdd-8b8c-42bddc69d528)

**Bug: get current game ở local storage, nếu current game null sao lấy được id** 
-> thêm ? vào

![image](https://github.com/FPTU-GHub-Capstone/ghub-app/assets/60615097/800e40a3-81e4-455a-b024-e3dd385c61af)

dg-panh commented:
Another bug: 

Clear local storage -> Login vào page -> click vào 1 game -> game level -> click vào create level

![image](https://github.com/FPTU-GHub-Capstone/ghub-app/assets/60615097/962d19b0-9aa2-4f27-b89f-30379ac89cfd)

Không get được current game ở local storage

![image](https://github.com/FPTU-GHub-Capstone/ghub-app/assets/60615097/f18ae6b0-31c6-4999-b7e0-e310bd1f198f)

_**thêm ? vào chứ để crash app thì kỳ**_

dg-panh commented:
![image](https://github.com/FPTU-GHub-Capstone/ghub-app/assets/60615097/672cff71-c129-459e-b4bd-ce423cdcb7d3)

Create game có ít field sao để dialog fullscreen vậy? nhìn hơi kỳ

thuanvmse161224 committed:
fix: fix localstorage error
+ change residue localhost:8080 to using variable
+ fix error when no localstorage in level create
+ fixing dialog size ( visual / UI )
+ remove snack, since already has another one running
+ use gameId params instead for the game create dialog

thuanvmse161224 committed:
Merge remote-tracking branch 'origin' into feature/gameCreate

dg-panh committed:
fix: 401 error

dg-panh committed:
feat: set private route

thuanvmse161224 committed:
feat: change logging layout, fix lint

thuanvmse161224 committed:
feat: close Logging on newtab

dg-panh committed:
feat: redirect game page

dg-panh committed:
Merge branch 'feature/gameCreate' of https://github.com/FPTU-GHub-Capstone/ghub-app into feature/gameCreate

thuanvmse161224 committed:
feat: adding FE form validate for gameCreate

dg-panh committed:
feat: use redux for game list
